Yank/repex.py

```diff
@@ -431,30 +433,36 @@ def write_mcmc_moves(self, mcmc_moves):
             serialized_move = mmtools.utils.serialize(mcmc_move)
             self.write_dict('mcmc_moves/move' + str(i), serialized_move)
 
-    def read_energies(self, iteration):
+    def read_energies(self, iteration=slice(None)):
         """Retrieve the energy matrix at the given iteration.
 
         Parameters
         ----------
-        iteration : int
-            The iteration at which to read the data.
+        iteration : int or slice
+            The iteration(s) at which to read the data. The slice(None) allows fetching all iterations at once.
 
         Returns
         -------
         energy_thermodynamic_states : n_replicas x n_replicas numpy.ndarray
-            energy_thermodynamic_states[i][j] is the reduced potential computed at
-            SamplerState sampler_states[i] and ThermodynamicState thermodynamic_states[j].
+            energy_thermodynamic_states[iteration, i, j] is the reduced potential computed at
+            SamplerState sampler_states[iteration, i] and ThermodynamicState thermodynamic_states[iteration, j].
         energy_unsampled_states : n_replicas x n_unsampled_states numpy.ndarray
-            energy_unsampled_states[i][j] is the reduced potential computed at SamplerState
-            sampler_states[i] and ThermodynamicState unsampled_thermodynamic_states[j].
+            energy_unsampled_states[iteration, i, j] is the reduced potential computed at SamplerState
```

Comment:
Yes, documentation will be tricky here since we can support a `slice(None)` and an int which changes the output